llvm/docs/LangRef.rst

```diff
@@ -1634,6 +1634,22 @@ Currently, only the following parameter attributes are defined:
 
     This attribute cannot be applied to return values.
 
+``range(<ty> <a>, <b>)``
+    This attribute expresses the possible range the parameter is in. The value of 
+    the parameter is in the specified range or is poison.
+    The arguments passed to ``range`` has the following properties:
+
+    -  The type must match the scalar type of the parameter.
```

Comment:
Is there any downside in having the range just apply to all vector elems?

Reply:
It does apply to all vector elements:

> For vector-typed parameters, the range is applied element-wise.

Reply:
oh I see, for vector types you pull out the element type when creating the range.